svg 로고 띄우기, 공통 버튼 컴포넌트 구현

thdwlsgus0 opened the pull request:
## 목표
 - hover 글자색이 초록색으로 바뀌는 버튼 구현
 - svg 파일로 로고를 만들어서 삽입

## 이슈 번호 
 - #4 #5 
## 체크 리스트
 - [o] 이미지 저장 버튼  
 - [p] export 버튼
 - [o] svg 파일로 로고를 만들기
 - [o] svg 파일을 리액트와 타입스크립트를 이용해서 웹 상에 띄우기

## 참고
      - 
<img width="297" alt="스크린샷 2020-11-24 오전 12 27 48" src="https://user-images.githubusercontent.com/22065725/99980722-e9529500-2deb-11eb-9636-14ddf192dfa9.png">

thdwlsgus0 committed:
feature : svg 추가

1. 로고를 만들었습니다.
2. declare를 이용해서 전역에서 svg를 정의하였습니다.
3. header에 로고를 추가하였습니다.

thdwlsgus0 committed:
modify : 파일 및 디렉토리 수정

1. EditableMathExample을 App으로 변경하였습니다.
2. header 디렉토리에 로고 이동하였습니다.

thdwlsgus0 committed:
feature : 공통 버튼 컴포넌트 생성

1. 공통 버튼 컴포넌트를 생성하였습니다.
2. hover시 글씨의 색상이 바뀌도록 구현하였습니다.

LeeSuKyeong commented on src/components/header/index.tsx:
```diff
@@ -0,0 +1,15 @@
+    <>
+      <SVGIcon />
+      <Button color="#fbfbfb" value="이미지 저장" />
+      <Button color="#B4F6E0" value="export" />
```
컬러 hex code 바로 쓰지말고 js파일로 만들어서 import 해오는게 더 깔끔할 것 같습니다.
ex) color.js
```
const color = {
 green: '#fbfbfb'
}
```

LeeSuKyeong replied:
' ' 로 지정한 줄 알았는데
" "로 된 거 봐선 eslint랑 prettier가 작동하지 않은걸가요???

thdwlsgus0 replied:
color를 파일로 빼서 관리하도록 하였습니다.

```
interface colorType {
  imageSave: string;
  exportSave: string;
}
const colors: colorType = {
  imageSave: '#fbfbfb',
  exportSave: '#B4F6E0',
};
export default colors;

```

zlrlo commented on package.json:
```diff
@@ -40,8 +41,12 @@
-    "react-mathquill": "^1.0.1"
+    "react-mathquill": "^1.0.1",
+    "styled-components": "^5.2.1",
+    "styled-normalize": "^8.0.7",
+    "styled-reset": "^4.3.1"
```
필요할 때 추가하는게 좋은거같아요!

thdwlsgus0 replied:
styled-reset은 global-style 설정할 떄 쓰려고 설치했습니다.
styled-normalize는 삭제하겠습니다.

zlrlo commented on src/utils/svg/logo.svg:
```diff
@@ -0,0 +1,23 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="100" height="50" viewBox="0 0 312 212">
+  <g id="사각형_1" data-name="사각형 1" fill="#23b6ac" stroke="#707070" stroke-width="1">
+    <rect width="312" height="212" rx="50" stroke="none"/>
+    <rect x="0.5" y="0.5" width="311" height="211" rx="49.5" fill="none"/>
+  </g>
+  <g id="패스_1" data-name="패스 1" transform="translate(18 19)" fill="#fff">
+    <path d="M 35 71.5 C 30.34407997131348 71.5 25.82755088806152 70.5621337890625 21.57588958740234 68.71244812011719 C 17.46806907653809 66.92534637451172 13.77853012084961 64.36660766601562 10.60976028442383 61.1072998046875 C 7.440040111541748 57.84701919555664 4.951379776000977 54.05031967163086 3.212899923324585 49.82266998291016 C 1.412750005722046 45.44503021240234 0.5 40.79441833496094 0 ...
+    <path d="M 35 1 C 30.41312026977539 1 25.96371078491211 1.923896789550781 21.77534866333008 3.746040344238281 C 17.72748184204102 5.507057189941406 14.09144973754883 8.02880859375 10.96826171875 11.24123001098633 C 7.843158721923828 14.45561981201172 5.389457702636719 18.19903945922852 3.675331115722656 22.36748886108398 C 1.900108337402344 26.68450164794922 1 31.27114105224609 1 36 C 1 40.72 ...
+  </g>
+  <g id="타원_2" data-name="타원 2" transform="translate(121 19)" fill="#fff" stroke="#fff" stroke-width="1">
+    <circle cx="36" cy="36" r="36" stroke="none"/>
+    <circle cx="36" cy="36" r="35.5" fill="none"/>
+  </g>
+  <g id="타원_3" data-name="타원 3" transform="translate(225 19)" fill="#fff" stroke="#fff" stroke-width="1">
+    <ellipse cx="34" cy="36" rx="34" ry="36" stroke="none"/>
+    <ellipse cx="34" cy="36" rx="33.5" ry="35.5" fill="none"/>
+  </g>
+  <text id="수식판" transform="translate(56 185)" fill="#fff" font-size="80" font-family="BMJUAOTF, BM JUA_OTF"><tspan x="0" y="0">수식판</tspan></text>
+  <line id="선_1" data-name="선 1" x2="256" y2="1" transform="translate(28.5 108.5)" fill="none" stroke="#fff" stroke-width="2"/>
+  <text id="_" data-name="+" transform="translate(37 71)" fill="#020202" font-size="60" font-family="BMJUAOTF, BM JUA_OTF"><tspan x="0" y="0">+</tspan></text>
+  <text id="_-" data-name="-" transform="translate(143 75)" fill="#020202" font-size="60" font-family="BMJUAOTF, BM JUA_OTF"><tspan x="0" y="0">-</tspan></text>
+  <text id="x" transform="translate(244 67)" fill="#020202" font-size="60" font-family="BMJUAOTF, BM JUA_OTF"><tspan x="0" y="0">x</tspan></text>
+</svg>
```
svg 따로 뺀거 좋은거 같아요!

thdwlsgus0 replied:
감사합니다 ㅎㅎ

zlrlo reviewed (commented):
수고하셨습니다 ~

thdwlsgus0 committed:
modify : 코드 수정

1. eslint 설정을 mac과 window에 맞도록 수정하였습니다.

thdwlsgus0 committed:
build : 필요없는 패키지 제거

1. 사용하지 않는 html-loader와 styled-normalize를 제거하였습니다.

thdwlsgus0 committed:
build : 사용하지 않는 패키지 제거

1. UglifyJsPlugin을 제거해주었습니다.
2. webpack의 html-loader부분을 제거해주었습니다.

thdwlsgus0 committed:
modify : 디렉토리 구조 변경 및 문법 변경

1. function -> 화살표 함수로 컴포넌트 구현하였습니다.
2. contexts-> stores, utils 내의 colors를 관리하도록 수정하였습니다.

thdwlsgus0 committed:
build : 필요없는 패키지 제거

1. 사용하지 않는 패키지를 제거하였습니다.

zlrlo commented on .eslintrc.js:
```diff
@@ -12,7 +12,12 @@ module.exports = {
   rules: {
     'import/no-unresolved': 'off',
-    'prettier/prettier': 1,
+    'prettier/prettier': [
+      'error',
+      {
+        endOfLine: 'auto',
+      },
+    ],
```
추가 감사합니다~

zlrlo commented on package.json:
```diff
@@ -24,7 +25,6 @@
-    "html-loader": "^1.3.2",
     "html-webpack-plugin": "^4.5.0",
     "mini-css-extract-plugin": "^1.3.1",
     "node-sass": "^5.0.0",
```
"node-sass" 이거 아직 안빠진거 같아요!

thdwlsgus0 replied:
다음 pr에 수정하겠습니다!!!

thdwlsgus0 replied:
수정해서 다시 pr에 추가하겠습니다/

zlrlo reviewed (commented):
감사합니다 ~
